ground/gcs/src/plugins/config/outputchannelform.cpp

```diff
@@ -387,9 +387,19 @@ void OutputChannelForm::updateMaxSpinboxValue(UAVObject *obj)
             // ... and if there's a match, set the maximum values and return
             if (channel-1 == m_index) {
                 if (actuatorSettingsData.TimerUpdateFreq[i] == 0)
+                {
+                    // SyncPWM has been selected, limit values to maximum allowed by UAVO
+                    ui.actuatorMin->setMaximum(std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max());
+                    ui.actuatorMax->setMaximum(std::numeric_limits<__typeof__(actuatorSettingsData.ChannelMax[0])>::max());
                     return;
-
-                double maxPulseWidth = round(10000000.0 / actuatorSettingsData.TimerUpdateFreq[i]);
+                }
+
+                double pwm_freq = 0;
+                if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_1MHZ)
+                    pwm_freq = 1000000;
+                else if (actuatorSettingsData.TimerPwmResolution[i] == ActuatorSettings::TIMERPWMRESOLUTION_12MHZ)
+                    pwm_freq = 12000000;
+                double maxPulseWidth = round(pwm_freq / actuatorSettingsData.TimerUpdateFreq[i]);
```

Comment:
I suppose ns vs. MHz is a matter of preference. But I used the 80ns value to calculate duty cycle for my application which resulted in an incorrect max pwm setting. I don't think it's a good idea to show incorrect values to the user.